From ac315f66be96c5a423382efb559f193e5b786226 Mon Sep 17 00:00:00 2001
From: Dave Stevenson <dave.stevenson@raspberrypi.com>
Date: Wed, 21 Oct 2020 18:33:02 +0100
Subject: [PATCH] dtoverlays: Correct CSI2 settings for ov9281

OV9281 appears to drop the clock to LP mode between frames, but
the overlay didn't define this at both ends of the CSI2 link.
The overlay also had an incorrect link frequency defined, not that
the driver ever checked for one.

Fix both issues.

Signed-off-by: Dave Stevenson <dave.stevenson@raspberrypi.com>
---
 arch/arm/boot/dts/overlays/ov9281-overlay.dts | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

--- a/arch/arm/boot/dts/overlays/ov9281-overlay.dts
+++ b/arch/arm/boot/dts/overlays/ov9281-overlay.dts
@@ -34,7 +34,7 @@
 						data-lanes = <1 2>;
 						clock-noncontinuous;
 						link-frequencies =
-							/bits/ 64 <456000000>;
+							/bits/ 64 <400000000>;
 					};
 				};
 			};
@@ -50,6 +50,7 @@
 				csi1_ep: endpoint {
 					remote-endpoint = <&ov9281_0>;
 					data-lanes = <1 2>;
+					clock-noncontinuous;
 				};
 			};
 		};
